site/book/02-concepts/02-package-lifecycle.md

```diff
@@ -0,0 +1,53 @@
+In this section, we'll describe the typical workflows in kpt. We say "typical", because there is no
+single right way of using kpt. A user may choose to use some command but not another. This
+modularity is a key design principle. However, we still want to provide guidance on how the
+functionality could be used in a end-to-end real-world scenario.
+
+A workflow in kpt can be best modelled as performing some verbs on the noun _package_.
+
+For example, when consuming an upstream package, the initial workflow can look like this:
+
+![img](/static/images/lifecycle/flow1.svg)
+
+- **Get**: Using `kpt pkg get`
+- **Explore**: Using an editor or running commands such as `kpt pkg tree`
+- **Customize**: Manual edits or using `kpt fn eval`
+- **Render**: Using `kpt fn render`
+
+First, you get a package from upstream. Then, you explore the content of the package to understand
+it better. Then you typically want to customize the package for you specific needs. Finally, you
+render the package which produces the final resources that can be directly applied to the cluster.
+
+This is an iterative process. There is usually a tight customize-render loop in order to produce
```

Comment:
If I run `kpt fn eval`, it renders the resources but for that function. `customize-render` is confusing. I think we should describe the difference using imperative and declarative concepts.

Reply:
I slightly rephrase this. I also renamed "Customize" to "Edit". The subtlety here is that you can edit a meta resource (Kptfile) either manually or automatically and then render the package. That's a form of customization.